src/main/io/serial_resource.c

```diff
@@ -0,0 +1,134 @@
+/*
+ * This file is part of Betaflight.
+ *
+ * Betaflight is free software. You can redistribute this software
+ * and/or modify this software under the terms of the GNU General
+ * Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * Betaflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "platform.h"
+
+#include "io/serial.h"
+
+// convert identifier into port type
+serialType_e serialType(serialPortIdentifier_e identifier)
+{
+#ifdef USE_VCP
+    if (identifier == SERIAL_PORT_USB_VCP) {
+        return SERIALTYPE_USB_VCP;
+    }
+#endif
+#ifdef USE_UART
+    if (identifier >= SERIAL_PORT_USART1 && identifier < SERIAL_PORT_USART1 + SERIAL_UART_MAX) {
+        unsigned idx = identifier - SERIAL_PORT_USART1;
+        if (BIT(idx) & SERIAL_UART_MASK) {
+            return SERIALTYPE_UART;
+        } else {
+            // no other type in this range
+            return SERIALTYPE_INVALID;
+        }
+    }
+#endif
+#ifdef USE_LPUART
```

Comment:
Note `USE_LPUART` and `USE_LPUART1` in this PR.

Reply:
All ports are normalized consistently. USE_LPUART1 means that LPUART1 is enabled, USE_LPUART means that any (but there is only LPUART1 now) is enabled.
I think it is used correctly.

Reply:
![Screenshot from 2024-10-30 18-27-54](https://github.com/user-attachments/assets/d71e2f7a-aa17-4694-9eab-d7a1f8b9bb44)

Please have a look why LPUART shows up for MATEKH743.

Reply:
fixed